### PR TITLE
[BEAM-3388] Use a typeswitch instead of reflect.Convert

### PR DESCRIPTION
**Pull request by lostluck**
Use a typeswitch instead of reflect.Convert when converting strings or bytes.

### CODE REVIEW

**Comment by lostluck** [comment]
R: @herohde @wcn3

File: sdks/go/pkg/beam/core/runtime/exec/coder.go [hunk 1]
```diff
@@ -134,9 +134,15 @@ type bytesEncoder struct{}
 
 func (*bytesEncoder) Encode(val FullValue, w io.Writer) error {
 	// Encoding: size (varint) + raw data
-
-	data := reflect.ValueOf(val.Elm).Convert(reflectx.ByteSlice).Interface().([]byte)
-	size := len(data)
```
Comment (herohde) [inline sdks/go/pkg/beam/core/runtime/exec/coder.go]: It seems this line got lost in editing. I don't think the code compiles. The Go presubmit test essentially runs "go test ./..." and should be green.

Comment (lostluck) [inline sdks/go/pkg/beam/core/runtime/exec/coder.go]: > **herohde** wrote:
> It seems this line got lost in editing. I don't think the code compiles. The Go presubmit test essentially runs "go test ./..." and should be green.


Good catch! Done.

**Comment by lostluck** [comment]
R: @aaltay (please merge) 
(It should work this time! Thank you!)

**Pull request merged**